src/otx/algo/visual_prompting/zero_shot_segment_anything.py

```diff
@@ -876,21 +878,36 @@ def initialize_reference_info(self) -> None:
         self.register_buffer("reference_feats", torch.zeros(0, 1, self.model.embed_dim), False)
         self.register_buffer("used_indices", torch.tensor([], dtype=torch.int64), False)
 
-    def _find_latest_reference_info(self, root: Path) -> str | None:
-        """Find latest reference info to be used."""
-        if not Path.is_dir(root):
-            return None
-        if len(stamps := sorted(os.listdir(root), reverse=True)) > 0:
-            return stamps[0]
-        return None
+    def save_reference_info(self, default_root_dir: Path | str) -> None:
+        """Save reference info."""
+        reference_info = {
+            "reference_feats": self.reference_feats,
+            "used_indices": self.used_indices,
+        }
+        # save reference info
+        path_reference_info: Path = Path(default_root_dir) / self.reference_info_dir / "reference_info.pt"
+        Path.mkdir(Path(path_reference_info).parent, parents=True, exist_ok=True)
```

Comment:
```suggestion
        path_reference_info.parent.mkdir(parents=True, exist_ok=True)
```
I think we can reduce using Path

Reply:
Done at https://github.com/openvinotoolkit/training_extensions/pull/3354/commits/d90ab5dfa60a1373f2a7ae26d99e87b139aa706e.